File: src/main/webapp/js/service/taskService.js
app.service('taskService',function($http){

    // 任务列表
    this.getAll=function(){
        return $http.get('../TaskController/getAll');
    }
    // 任务列表
    this.getOne=function(taskId){
        return $http.get('../TaskController/getOne?taskId='+taskId);
    }
    // 任务列表
    this.searchByUsername=function(username){
        return $http.get('../TaskController/searchByUsername?username='+username);
    }
    // 任务列表
    this.searchByProjectId=function(projectId){
        return $http.get('../TaskController/searchByProjectId?projectId='+projectId);
    }
    // 新建任务
    this.insert=function(entity){
        return $http.post('../TaskController/insert',entity);
    }
    // 删除任务
    this.dele=function(selectIds){
        if(confirm("你确认要删除么")) {
            return $http.get('../TaskController/delete?ids=' + selectIds);
        }
    }
    // 更新任务信息
    this.update=function(entity){
        return $http.post('../TaskController/update',entity);
    }
})

File: src/main/java/com/talenco/tasksystem/service/TaskService.java
package com.talenco.tasksystem.service;

import com.talenco.tasksystem.entity.Task;

import java.util.List;

public interface TaskService {

    List<Task> getAll();

    Task getOne(Long id);

    Long insert(Task task);

    void update(Task task);

    /**
     * 删除任务
     * @param ids
     */
    void delete(Long[] ids);

    List<Task> searchByProjectId(Long projectId);

    List<Task> searchByUsername(String username);
}
